core/data/src/test/java/uk/gov/gchq/gaffer/data/element/function/ReduceRelatedElementsTest.java

```diff
@@ -0,0 +1,453 @@
+
+        // Then
+        assertElementEquals(elements, result);
+    }
+
+    @Test
+    public void shouldFailIfNoVisibilityAggregatorProvided() {
+        Object vertex1a = "vertex1a longest";
+        Object vertex1b = "vertex1b";
+        Object vertex2a = "vertex2a";
+        Object vertex2b = "vertex2b longest";
+        Object vertex3a = "vertex3a";
+        Object vertex3b = "vertex3b longest";
+
+        // Given
+        final ReduceRelatedElements function = new ReduceRelatedElements();
+        function.setVisibilityProperty(VISIBILITY);
+        function.setVisibilityAggregator(null);
+        function.setVertexAggregator(new Longest());
+        function.setRelatedVertexGroups(Collections.singleton(RELATES_TO));
+
+        final List<Element> elements = Arrays.asList(
+                new Edge.Builder()
+                        .source(vertex1a)
+                        .dest(vertex2a)
+                        .directed(true)
+                        .group(TestGroups.EDGE)
+                        .property(VISIBILITY, Sets.newHashSet("public"))
+                        .build(),
+                new Edge.Builder()
+                        .source(vertex1b)
+                        .dest(vertex3a)
+                        .directed(true)
+                        .group(TestGroups.EDGE)
+                        .property(VISIBILITY, Sets.newHashSet("public"))
+                        .build(),
+                new Entity.Builder()
+                        .vertex(vertex2a)
+                        .group(TestGroups.ENTITY)
+                        .property(VISIBILITY, Sets.newHashSet("public"))
+                        .build(),
+                new Edge.Builder()
+                        .source(vertex1a)
+                        .dest(vertex1b)
+                        .directed(true)
+                        .group(RELATES_TO)
+                        .property(VISIBILITY, Sets.newHashSet("public"))
+                        .build(),
+                new Edge.Builder()
+                        .source(vertex2a)
+                        .dest(vertex2b)
+                        .directed(true)
+                        .group(RELATES_TO)
+                        .property(VISIBILITY, Sets.newHashSet("public"))
+                        .build(),
+                new Edge.Builder()
+                        .source(vertex3a)
+                        .dest(vertex3b)
+                        .directed(true)
+                        .group(RELATES_TO)
+                        .property(VISIBILITY, Sets.newHashSet("private"))
+                        .build()
+        );
+
+        // When
+        try {
+            function.apply(elements);
+
+            fail();
+        } catch (final IllegalArgumentException e) {
+            // Then
+            assertTrue(e.getMessage().contains("No visibility aggregator provided, so visibilities cannot be combined."));
+        }
+    }
+
```

Comment:
It might be worth including some more granular testing of the (currently private) logic found in some of the helper methods.

Reply:
Might be worth extracting the updater method to its own class/set of helper methods to keep encapsulation.